### ISSUE TITLE
Breaks Atom

### ISSUE THREAD

**Issue by Sod-Almighty** [morassman/tabasco#10]
Do you realise that your package breaks Atom? I just spent an HOUR disabling packages and reloading Atom trying to track down the problem. Thanks a bunch.

**Atom**: 1.25.0 x64
**Electron**: 1.7.11
**OS**: Mac OS X 10.11.6
**Thrown From**: [tabasco](https://github.com/morassman/tabasco) package 0.5.0

### Stack Trace

Failed to activate the tabasco package

```
At tabBarView.addEventListener is not a function

TypeError: tabBarView.addEventListener is not a function
    at Object.tabBarViewAdded (/packages/tabasco/lib/tabasco.coffee:112:16)
    at Object.connectPaneWithTabBarView (/packages/tabasco/lib/tabasco.coffee:80:10)
    at Object.paneAdded (/packages/tabasco/lib/tabasco.coffee:51:10)
    at /packages/tabasco/lib/tabasco.coffee:36:65
    at PaneContainer.observePanes (/Applications/Atom.app/Contents/Resources/app/src/pane-container.js:78:47)
    at WorkspaceCenter.observePanes (/Applications/Atom.app/Contents/Resources/app/src/workspace-center.js:162:37)
    at CompositeDisposable.getPaneContainers.map.container (/Applications/Atom.app/Contents/Resources/app/src/workspace.js:794:68)
    at Array.map (native)
    at Workspace.observePanes (/Applications/Atom.app/Contents/Resources/app/src/workspace.js:794:41)
    at Object.tabsActivated (/packages/tabasco/lib/tabasco.coffee:36:41)
    at Object.activate (/packages/tabasco/lib/tabasco.coffee:20:8)
    at Package.activateNow (/Applications/Atom.app/Contents/Resources/app/src/package.js:230:33)
    at measure (/Applications/Atom.app/Contents/Resources/app/src/package.js:206:33)
    at Package.measure (/Applications/Atom.app/Contents/Resources/app/src/package.js:88:25)
    at activationPromise.Promise (/Applications/Atom.app/Contents/Resources/app/src/package.js:200:20)
    at Promise (<anonymous>)
    at Package.activate (/Applications/Atom.app/Contents/Resources/app/src/package.js:198:38)
    at PackageManager.activatePackage (/Applications/Atom.app/Contents/Resources/app/src/package-manager.js:695:42)
    at packagesToEnable.forEach.name (/Applications/Atom.app/Contents/Resources/app/src/package-manager.js:440:51)
    at Array.forEach (native)
    at disabledPackagesSubscription.config.onDidChange (/Applications/Atom.app/Contents/Resources/app/src/package-manager.js:440:30)
    at emitter.on (/Applications/Atom.app/Contents/Resources/app/src/config.js:1023:22)
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:25:20)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app/node_modules/event-kit/lib/emitter.js:141:34)
    at Config.emitChangeEvent (/Applications/Atom.app/Contents/Resources/app/src/config.js:1175:62)
    at Config.setRawValue (/Applications/Atom.app/Contents/Resources/app/src/config.js:1008:23)
    at Config.set (/Applications/Atom.app/Contents/Resources/app/src/config.js:711:18)
    at Config.removeAtKeyPath (/Applications/Atom.app/Contents/Resources/app/src/config.js:901:16)
    at Package.enable (/Applications/Atom.app/Contents/Resources/app/src/package.js:75:30)
    at PackageManager.enablePackage (/Applications/Atom.app/Contents/Resources/app/src/package-manager.js:246:18)
```

### Commands

```
  4x -3:41.3.0 core:backspace (input.hidden-input)
     -0:43.4.0 pane:close (section.section)
     -0:32.3.0 command-palette:toggle (div.panels-item)
  6x -0:30 core:backspace (input.hidden-input)
     -0:26.2.0 editor:select-to-first-character-of-line (input.hidden-input)
     -0:23.3.0 editor:consolidate-selections (input.hidden-input)
     -0:23.3.0 core:cancel (input.hidden-input)
```

### Non-Core Packages

```
arpeggio 1.1.2 
atom-brackets-glow 0.6.0 
atom-ctags 5.1.2 
atom-like-brackets-editor 1.2.0 
atom-material-syntax 1.0.8 
atom-material-ui 2.1.3 
autocomplete-clang 0.11.4 
autocomplete-ruby 0.2.8 
busy-signal 1.4.3 
color-picker 2.2.5 
custom-folds 1.9.5 
custom-invisibles 0.1.2 
export-html 3.1.0 
file-icons 2.1.18 
file-types 0.5.5 
fold-comments 0.6.0 
fold-functions 0.5.0 
fold-lines 0.2.2 
fun-files 0.10.0 
grammar-token-limit 0.1.1 
highlight-bad-chars 1.0.7 
highlight-column 0.5.4 
indent-detective 0.2.0 
indent-guide-improved 1.4.13 
intentions 1.1.5 
jumpy 4.2.0 
language-dylan 0.7.3 
language-haml 0.31.0 
language-ini 1.19.0 
language-lisp 0.2.0 
language-livescript 0.0.3 
language-ruby-without-erb 0.60.0 
language-scheme 1.4.0 
line-ending-converter 1.3.2 
linter 2.2.0 
linter-clang 4.1.2 
linter-gcc 0.7.1 
linter-ruby 1.2.6 
linter-ui-default 1.7.1 
multi-cursor-plus 1.2.0 
navigation-history 0.1.8 
Parinfer 1.18.0 
periodic 0.2.3 
permanent-delete 0.1.7 
pigments 0.40.2 
project-folder 1.5.0 
project-plus 1.0.0 
rainbow-delimiters 2.1.2 
rainbow-tabs 2.5.1 
regex-railroad-diagram 0.19.4 
remember-folds 0.3.0 
remote-sync 4.1.8 
ruby-block 0.3.6 
ruby-block-converter 4.0.1 
ruby-navigator 0.5.0 
show-invisibles-plus 0.3.1 
simple-drag-drop-text 0.4.0 
Stylus 3.1.1 
swackets 0.35.0 
switch-header-source 0.27.2 
sync-settings 0.8.6 
tab-switcher 1.5.6 
tabasco 0.5.0 
tabs-to-spaces 1.0.3 
terminal-plus 0.14.5 
text-pastry 0.6.1 
toggle-gutter 0.3.4 
tool-bar 1.1.0 
trailing-spaces 0.4.0 
tree-ignore 0.6.2 
tree-view-breadcrumb 0.8.1 
tree-view-filter 0.5.3

**Comment by morassman** [comment]
When did you install this package? It has been unpublished from Atom a long time ago already. It shouldn't even be possible to install it anymore. In fact. If you search for it either from within Atom or the packages section on the site it won't even show up anymore.

I assume you installed it a long time ago and it only breaks now perhaps due to the latest Atom update. The ability to drag and dock tabs was implemented in Atom after tabasco was created, which now makes tabasco obsolete. Just uninstall it.

PS. Please try to refrain from sarcasm or any form of hostility.

> Thanks a bunch.

A lot of one's free time and effort goes into making these packages. When logging an issue keep that in mind and give the author the benefit of the doubt.

**Comment by Sod-Almighty** [comment]
If it's no longer relevant, then Atom should tell me. It shouldn't be my responsibility to magically know that an add-on I've been using for ages is suddenly incompatible and breaks Atom.

It is the responsibility of the add-on to check (or at least define) its dependencies. It is then the responsibility of the add-on or the host application to display errors if these dependencies are not met.

Ruby gems, for example, have a dependency file that specifies what versions of Ruby and other gems are required. If I update Ruby to a later version than a gem requires, it will throw a meaningful error.

Your add-on, by comparison, responds to the incompatibility by **breaking Atom** rather than informing the user of the problem. Therefore either your add-on - or Atom itself - has wasted my time by not checking dependencies. Therefore I am perfectly within my rights to be _slightly_ sarcastic when my valuable time is wasted. Trust me, I could *easily* have been more "hostile".

In short, if the add-on is no longer relevant - and, indeed, breaks Atom - why does it continue to run? I repeat: It is not the responsibility of the end user to magically know when an add-on stops working. I have been using this add-on for months, so the only way I would know it's obsolete is *if it tells me!*

Therefore either *you*, or the developers of Atom, have wasted my time; depending on whether Atom provides any way for an add-on to check for its own obsolescence.

**Comment by morassman** [comment]
You are assuming that solving the problem is as simple as checking the dependencies. Even a package that meets all the dependencies can cause Atom to crash.

In the open source community one cannot have a sense of entitlement such as this. If you want to have the liberty to do something "within your rights", then rather use a paid product. Logging a bug on an open source project should be for the purpose of improving the project. Not to accuse and attack the author.

I'd encourage you to familiarise yourself with Github's terms of service and code of conduct. Abuse like this is not tolerated.

**Comment by Sod-Almighty** [comment]
Then you have a very odd definition of "abuse".
